tests/unit/project_loader/extensions/test_flutter_master.py

```diff
@@ -0,0 +1,116 @@
+
+import textwrap
+
+from testtools.matchers import Equals
+
+from snapcraft.internal.project_loader._extensions.flutter_master import ExtensionImpl
+
+from .. import ProjectLoaderBaseTest
+
+
+class ExtensionTest(ProjectLoaderBaseTest):
+    def test_extension(self):
+        flutter_extension = ExtensionImpl(
+            extension_name="flutter-master", yaml_data=dict(base="core18")
+        )
+
+        self.expectThat(
+            flutter_extension.root_snippet,
+            Equals(
+                {
+                    "plugs": {
+                        "gtk-3-themes": {
+                            "interface": "content",
+                            "target": "$SNAP/data-dir/themes",
+                            "default-provider": "gtk-common-themes",
+                        },
+                        "icon-themes": {
+                            "interface": "content",
+                            "target": "$SNAP/data-dir/icons",
+                            "default-provider": "gtk-common-themes",
+                        },
+                        "sound-themes": {
+                            "interface": "content",
+                            "target": "$SNAP/data-dir/sounds",
+                            "default-provider": "gtk-common-themes",
+                        },
+                        "gnome-3-28-1804": {
+                            "interface": "content",
+                            "target": "$SNAP/gnome-platform",
+                            "default-provider": "gnome-3-28-1804",
+                        },
+                    },
+                    "environment": {"SNAP_DESKTOP_RUNTIME": "$SNAP/gnome-platform"},
+                    "layout": {
+                        "/usr/share/xml/iso-codes": {
+                            "bind": "$SNAP/gnome-platform/usr/share/xml/iso-codes"
+                        },
+                    },
+                }
+            ),
+        )
+        self.expectThat(
+            flutter_extension.app_snippet,
+            Equals(
+                {
+                    "command-chain": ["snap/command-chain/desktop-launch"],
+                    "plugs": [
+                        "desktop",
+                        "desktop-legacy",
+                        "gsettings",
+                        "opengl",
+                        "wayland",
+                        "x11",
+                    ],
+                }
+            ),
+        )
+        self.expectThat(flutter_extension.part_snippet, Equals(dict()))
+        self.expectThat(
+            flutter_extension.parts,
+            Equals(
+                {
+                    "gnome-3-28-extension": {
+                        "source": "$SNAPCRAFT_EXTENSIONS_DIR/desktop",
+                        "source-subdir": "gnome",
+                        "plugin": "make",
+                        "build-packages": ["gcc", "libgtk-3-dev"],
+                    },
+                    "flutter-extension": {
```

Comment:
Include full name of extension? s/"flutter-extension"/"flutter-master-extension"

Reply:
I would prefer to keep it like this here, just like for gnome and for good reason, we cannot use them together